rc/filetype/javascript.kak

```diff
@@ -55,21 +55,112 @@ define-command -hidden javascript-indent-on-char %<
-        # deindent closing token(s) when after cursor
-        try %_ execute-keys -draft <a-x> <a-k> ^\h*[})\]] <ret> gh / [})\]] <ret> m <a-S> 1<a-&> _
+    execute-keys <semicolon>
+    try %<
+        # if previous line is part of a comment, do nothing
+        execute-keys -draft <a-?>/\*<ret> <a-K>^\h*[^/*\h]<ret>
+    > catch %<
+        # else if previous line closed a paren (possibly followed by words and a comment),
+        # copy indent of the opening paren line
+        execute-keys -draft k<a-x> 1s(\))(\h+\w+)*\h*(\;\h*)?(?://[^\n]+)?\n\z<ret> m<a-semicolon>J <a-S> 1<a-&>
+    > catch %<
+        # else indent new lines with the same level as the previous one
+        execute-keys -draft K <a-&>
+    >
+    # remove previous empty lines resulting from the automatic indent
+    try %< execute-keys -draft k <a-x> <a-k>^\h+$<ret> Hd >
+    # indent after an opening brace or parenthesis at end of line
+    try %< execute-keys -draft k <a-x> <a-k>[{(]\h*$<ret> j <a-gt> >
+    # indent after a label (works for case statements)
+    try %< execute-keys -draft k <a-x> s[a-zA-Z0-9_-]+:\h*$<ret> j <a-gt> >
+    # indent after a statement not followed by an opening brace
+    try %< execute-keys -draft k <a-x> s\)\h*(?://[^\n]+)?\n\z<ret> \
+                               <a-semicolon>mB <a-k>\A\b(if|for|while)\b<ret> <a-semicolon>j <a-gt> >
+    try %< execute-keys -draft k <a-x> s \belse\b\h*(?://[^\n]+)?\n\z<ret> \
+                               j <a-gt> >
+    # deindent after a single line statement end
+    try %< execute-keys -draft K <a-x> <a-k>\;\h*(//[^\n]+)?$<ret> \
+                               K <a-x> s\)(\h+\w+)*\h*(//[^\n]+)?\n([^\n]*\n){2}\z<ret> \
+                               MB <a-k>\A\b(if|for|while)\b<ret> <a-S>1<a-&> >
+    try %< execute-keys -draft K <a-x> <a-k>\;\h*(//[^\n]+)?$<ret> \
+                               K <a-x> s \belse\b\h*(?://[^\n]+)?\n([^\n]*\n){2}\z<ret> \
+                               <a-S>1<a-&> >
+    # deindent closing brace(s) when after cursor
+    try %< execute-keys -draft <a-x> <a-k> ^\h*[})] <ret> gh / [})] <esc> m <a-S> 1<a-&> >
+    # align to the opening parenthesis or opening brace (whichever is first)
+    # on a previous line if its followed by text on the same line
+    try %< evaluate-commands -draft %<
+        # Go to opening parenthesis and opening brace, then select the most nested one
+        try %< execute-keys [c [({],[)}] <ret> >
+        # Validate selection and get first and last char
+        execute-keys <a-k>\A[{(](\h*\S+)+\n<ret> <a-K>"(([^"]*"){2})*<ret> <a-K>'(([^']*'){2})*<ret> <a-:><a-semicolon>L <a-S>
+        # Remove possibly incorrect indent from new line which was copied from previous line
+        try %< execute-keys -draft <space> <a-h> s\h+<ret> d >
+        # Now indent and align that new line with the opening parenthesis/brace
+        execute-keys 1<a-&> &
+     > >
+        # # preserve previous line indent
```

Comment:
this block as well

Reply:
same, removed -- commenting works as expected now.